nltk/test/unit/test_tokenize.py

```diff
@@ -470,3 +462,51 @@ def test_punkt_tokenize_no_custom_lang_vars(self):
         expected = ["উপরাষ্ট্রপতি শ্রী এম ভেঙ্কাইয়া নাইডু সোমবার আই আই টি দিল্লির হীরক জয়ন্তী উদযাপনের উদ্বোধন করেছেন। অনলাইনের মাধ্যমে এই অনুষ্ঠানে কেন্দ্রীয় মানব সম্পদ উন্নয়নমন্ত্রী শ্রী রমেশ পোখরিয়াল ‘নিশাঙ্ক’  উপস্থিত ছিলেন। এই উপলক্ষ্যে উপরাষ্ট্রপতি হীরকজয়ন্তীর লোগো এবং ২০৩০-এর জন্য প্রতিষ্ঠানের লক্ষ্য ও পরিকল্পনার নথি প্রকাশ করেছেন।"]
 
         assert obj.tokenize(sentences) == expected
+
+    def test_punkt_debug_decisions_dot(self):
+        # Test debug_decisions on a text with two sentences, split by a dot.
+        tokenizer = punkt.PunktSentenceTokenizer()
+        txt = "Subject: Some subject. Attachments: Some attachemnts"
+
+        # Test that the sentence is split into two sections,
+        # with one split and hence one decision.
+        assert len(tokenizer.tokenize(txt)) == 2
+        assert len(list(tokenizer.debug_decisions(txt))) == 1
+
+    def test_punkt_debug_decisions_exclamation_mark(self):
+        # Test debug_decisions on a text with two sentences, split by a dot.
+        tokenizer = punkt.PunktSentenceTokenizer()
+        txt = "Subject: Some subject! Attachments: Some attachemnts"
+
+        # Test that the sentence is split into two sections,
+        # with one split and hence one decision.
+        assert len(tokenizer.tokenize(txt)) == 2
+        assert len(list(tokenizer.debug_decisions(txt))) == 1
+
+    def test_punkt_debug_decisions_custom_end(self):
+        # Test debug_decisions on a text with two sentences,
+        # split by a custom end character, based on Issue #2519
+        tokenizer = punkt.PunktSentenceTokenizer()
+
+        class ExtLangVars(punkt.PunktLanguageVars):
+            sent_end_chars = ('.', '?', '!', '^')
+
+        tokenizer._lang_vars = ExtLangVars()
+
+        txt = "Subject: Some subject^ Attachments: Some attachemnts"
+
+        # Test that the sentence is split into two sections,
+        # with one split and hence one decision.
+        assert len(tokenizer.tokenize(txt)) == 2
+        assert len(list(tokenizer.debug_decisions(txt))) == 1
+
+    def test_punkt_debug_decisions_no_split(self):
```

Comment:
optional: you could do `test_punkt_debug_decisions_no_split`, `test_punkt_debug_decisions_exclamation_mark`, and `test_punkt_debug_decisions_dot` as one test with `pytest.mark.parametrize`

Reply:
I've done so! Thank you for the reminder and recommendation.